### PR TITLE
chore: migrate to poetry

### PR DESCRIPTION
**Pull request by cyclimse**
## Description

Migrate the project to poetry to avoid redundancy between the requirements.

Changes:
- switch from setuptools to poetry
- remove the requirements files
- add a readthedocs.yml so that readthedocs use the dependencies from poetry
- update the ci to use poetry

### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request.
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

- [ ] I have added unit test covering every changes I have made
- [ ] I have updated the relevant documentation

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -44,7 +44,3 @@ repos:
     hooks:
       - id: python-bandit-vulnerability-check
         args: [--skip, "B101", --recursive, clumper]
-  - repo: https://github.com/regebro/pyroma
-    rev: "4.1"
-    hooks:
-      - id: pyroma
```
Comment (cyclimse) [inline .pre-commit-config.yaml:50]: Sadly pyroma is incompatible with poetry AFAIK.

File: .github/actions/setup-dev/action.yml [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+    - uses: actions/cache@v3
+      id: npm-cache
+      with:
+        path: ~/.npm
+        key: ${{ runner.os }}-build-${{ env.cache-name }}
+
+    - name: Install npm dependencies
+      shell: bash
+      run: npm install -g serverless serverless-scaleway-functions
```
Comment (Shillaker) [inline .github/actions/setup-dev/action.yml:19]: Can we split out the `npm`-related stuff into a separate action? It's only necessary for the SF integration tests I think.

**Review by Shillaker — CHANGES_REQUESTED** [review]

File: .github/actions/setup-dev/action.yml (hunk 2 above)
Comment (cyclimse) [reply .github/actions/setup-dev/action.yml]: Excellent, thanks! Same goes with secrets :face_with_spiral_eyes: I was mounting them but not using them.

File: .github/workflows/pytest-integration.yml [hunk 3]
```diff
@@ -15,18 +15,10 @@ jobs:
...
-          if [ -f requirements-dev.txt ]; then pip install -r requirements-dev.txt; fi
-          npm install -g serverless serverless-scaleway-functions
+      - uses: ./.github/actions/setup-dev
       - name: Test with pytest
         working-directory: tests
-        run: pytest integrations -n $(nproc --all)
+        run: poetry run pytest integrations -n $(nproc --all)
         env:
           SCW_DEFAULT_ORGANIZATION_ID: ${{ secrets.SCW_DEFAULT_ORGANIZATION_ID }}
           SCW_SECRET_KEY: ${{ secrets.SCW_SECRET_KEY }}
```
[the remark this answers is not in the document]
Comment (cyclimse) [reply .github/workflows/pytest-integration.yml:33]: What do you mean ?

**Review by thibaultjunin — APPROVED** [review]

**Review by Shillaker — APPROVED** [review]

**Pull request merged**